### ISSUE TITLE
[clang] Designated Initializer + Anonymous Union + Generic Lambda false positive assignment bug

### ISSUE THREAD

**Issue by rfenelus** [llvm/llvm-project#62023]
This combination of a designated initializers, an anonymous union, and a generic lambda does not compile in Clang but does compile in GCC and MSVC.  Replacing the assignment value with an expression that doesn't include a variable seems to work but breaks with variables.

```c++
int main(){
	struct Value{
		union{
			double value;
		};
	};

	auto Lambda = [](auto self) -> Value{
		double v  = 0;
		return Value{.value = v};
	};

	Lambda (Lambda);
}
```
[Godbolt](https://godbolt.org/z/YnaWzcha4)

**Comment by shafik** [comment]
Confirmed, this looks like: https://github.com/llvm/llvm-project/issues/57436

**Comment by Fznamznon** [comment]
I think this is fixed by https://github.com/llvm/llvm-project/pull/69712 , online demo - https://godbolt.org/z/bT1s85dfe .
